libs/shared/ui/src/lib/components/link/link.tsx

```diff
@@ -1,58 +1,92 @@
-import { NavLink } from 'react-router-dom'
-import { type IconEnum } from '@qovery/shared/enums'
+import { type VariantProps, cva } from 'class-variance-authority'
+import { type ComponentPropsWithoutRef, forwardRef } from 'react'
+import { NavLink, type NavLinkProps } from 'react-router-dom'
+import { twMerge } from 'tailwind-merge'
 import Icon from '../icon/icon'
+import { IconAwesomeEnum } from '../icon/icon-awesome.enum'
 
+// TODO: already used on several components need to be removed
 export interface BaseLink {
   link: string
   linkLabel?: string
   external?: boolean
 }
 
-export interface LinkProps extends BaseLink {
-  className?: string
-  size?: string
-  iconRight?: IconEnum | string
-  iconRightClassName?: string
-  iconLeft?: IconEnum | string
-  iconLeftClassName?: string
-}
+const linkVariants = cva(
+  ['cursor-pointer', 'transition', 'duration-100', 'font-medium', 'inline-flex', 'flex-center', 'gap-1'],
+  {
+    variants: {
+      color: {
+        brand: ['text-brand-500', 'hover:text-brand-600'],
+        red: ['text-red-500', 'hover:text-red-600'],
+        sky: ['text-sky-500', 'hover:text-sky-600'],
+      },
+      size: {
+        xs: ['text-xs'],
+        ssm: ['text-ssm'],
```

Comment:
This size naming seems weird.
I think we want to expose only `xs` `sm`, `md`, even if it resolves to `text-ssm` internally.
I'd like to remove the `text-ssm` on the long run.

Reply:
I used the same that our Tailwind sizes setup, it should be easier to use if we have the same "name".
For your point, I understand, but we'll need refactoring of all font sizes if we want it